src/climatevision/generator/industry2018/energy_source/__init__.py

```diff
@@ -4,109 +4,151 @@
-        + renew_solarth.energy
-        + renew_elec.energy
-    )
-
-    total = EnergyWithPercentage(
-        energy=entries.i_energy_total, total_energy=entries.i_energy_total
-    )
-
-    return EnergySupply(
-        total=total,
-        fossil=fossil,
-        fossil_gas=fossil_gas,
-        fossil_coal=fossil_coal,
-        fossil_diesel=fossil_diesel,
-        fossil_fueloil=fossil_fueloil,
-        fossil_lpg=fossil_lpg,
-        fossil_opetpro=fossil_opetpro,
-        fossil_ofossil=fossil_ofossil,
-        renew=renew,
-        renew_biomass=renew_biomass,
-        renew_heatnet=renew_heatnet,
-        renew_heatpump=renew_heatpump,
-        renew_solarth=renew_solarth,
-        renew_elec=renew_elec,
+class EnergySource:
+    s: EnergyWithPercentage
+    s_fossil: Energy
+    s_fossil_gas: EnergyWithPercentage
+    s_fossil_coal: EnergyWithPercentage
+    s_fossil_diesel: EnergyWithPercentage
+    s_fossil_fueloil: EnergyWithPercentage
+    s_fossil_lpg: EnergyWithPercentage
+    s_fossil_opetpro: EnergyWithPercentage
+    s_fossil_ofossil: EnergyWithPercentage
+    s_renew: Energy
+    s_renew_biomass: EnergyWithPercentage
+    s_renew_heatnet: EnergyWithPercentage
+    s_renew_elec: EnergyWithPercentage
+    s_renew_orenew: EnergyWithPercentage
+
+
+# TODO Energy calculation with new facts for each sector
+def calc_supply(inputs: Inputs, production: Production) -> EnergySource:
+
+    # do for all subbranches + copy structure for branches from production_branches
+    s_miner_cement = EnergySourceSubBranch.calc_energy_source_sub_branch(
+        inputs=inputs,
+        energy_demand=production.miner_cement.energy,
+        sub_branch="cement",
+        branch="miner",
+    )
+    s_miner_chalk = EnergySourceSubBranch.calc_energy_source_sub_branch(
+        inputs=inputs,
+        energy_demand=production.miner_chalk.energy,
+        sub_branch="chalk",
+        branch="miner",
+    )
+    s_miner_glas = EnergySourceSubBranch.calc_energy_source_sub_branch(
+        inputs=inputs,
+        energy_demand=production.miner_glas.energy,
+        sub_branch="glas",
+        branch="miner",
+    )
+    s_miner_ceram = EnergySourceSubBranch.calc_energy_source_sub_branch(
+        inputs=inputs,
+        energy_demand=production.miner_ceram.energy,
+        sub_branch="ceram",
+        branch="miner",
+    )
+    s_miner = EnergySourceBranch.calc_energy_source_sum(
+        sub_branch_list=[s_miner_cement, s_miner_chalk, s_miner_glas, s_miner_ceram]
+    )
+
+    s_chem_basic = EnergySourceSubBranch.calc_energy_source_sub_branch(
+        inputs=inputs,
+        energy_demand=production.chem_basic.energy,
+        sub_branch="basic",
+        branch="chem",
+    )
+    s_chem_ammonia = EnergySourceSubBranch.calc_energy_source_sub_branch(
+        inputs=inputs,
+        energy_demand=production.chem_ammonia.energy,
+        sub_branch="basic",  # assumtion same as chem basic (TODO Find specific factors for ammonia production)
```

Comment:
@JanLK I assume this is a todo for later and other than fixing the spelling of assumption we should leave this as is for now?

Reply:
This is a todo to improve the calculation, as right now the factors for basic chemical industry are used for the calculation of the ammonia sector.
In my opinion this minor uncetrtainty is not preventing us to merge.
The two options are:
- add facts for ammonia (with values from chem basic) and change those later
- leave as is and change this code and add the facts in one go